### PR TITLE
consider ValueChange triggers on path /* as invalid

### PR DESCRIPTION
**Pull request by Pavinati**
Show error feedback on data trigger path when trigger is
ValueChange and path /*
See also https://github.com/astarte-platform/astarte/issues/513

Signed-off-by: Mattia <mattia.pavinati@ispirata.com>

### CODE REVIEW

File: src/elm/Page/TriggerBuilder.elm [hunk 1]
```diff
@@ -1169,6 +1169,22 @@ triggerEventOptions currentEvent isPropertyInterface =
...
+        isAny =
+            path == "/*"
+
+        isValueChangeEvent =
+            event == DataTrigger.ValueChange || event == DataTrigger.ValueChangeApplied
+
+        matchesInterface =
+            mappingType /= Nothing
+    in
+    -- TODO this is a workaround, reenable ValueChange after being fixed in Astarte
```
Comment (bettio) [inline src/elm/Page/TriggerBuilder.elm:1184]: re-enable

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Changed
 - Disable ValueChanged, ValueChangedApplied and PathCreated triggers
   on datastream interfaces
+- Disable ValueChanged, ValueChangedApplied triggers on /* paths
```
Comment (bettio) [inline CHANGELOG.md:11]: ... (workaround to ref-to-issue-here)

**Review by bettio — CHANGES_REQUESTED** [review]

**Review by bettio — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Changed
 - Disable ValueChanged, ValueChangedApplied and PathCreated triggers
   on datastream interfaces
+- Disable ValueChanged and ValueChangedApplied triggers on /* paths
```
Comment (rbino) [inline CHANGELOG.md:11]: Typo: ValueChange and ValueChangeApplied

File: src/elm/Page/TriggerBuilder.elm [hunk 4]
```diff
@@ -1169,6 +1169,22 @@ triggerEventOptions currentEvent isPropertyInterface =
...
+isValidPath : String -> DataTriggerEvent -> Maybe MappingType -> Bool
+isValidPath path event mappingType =
+    let
+        isAny =
+            path == "/*"
+
+        isValueChangeEvent =
+            event == DataTrigger.ValueChange || event == DataTrigger.ValueChangeApplied
+
+        matchesInterface =
```
Comment (rbino) [inline src/elm/Page/TriggerBuilder.elm:1181]: I'm not sure about the name of this variable, maybe `matchesMapping` is better?

**Review by rbino — CHANGES_REQUESTED** [review]

**Review by rbino — APPROVED** [review]

**Review by bettio — APPROVED** [review]

**Pull request merged**